homeworks/vancsikpeti/5_oop/car_fleet_mgr.py

```diff
@@ -0,0 +1,114 @@
+class Car:
+    total_distance = 0.0
+
+    def __init__(self, brand: str, model: str, year: int, mileage = 0, fuel_level = 100):
+        self.brand = brand
+        self.model = model
+        self.year = year
+        self.mileage = mileage # induló értéke 0
+        self.fuel_level = fuel_level # induló értéke 100, tételezzük fel, hogy 0.1% üzemanyag fogy megtett kilométerenként
+   
+    #def new_car(self):
+        #self.brand = input("Brand of car: ")
+        #self.model = input("Model of car: ")
+        #self.year = input("Product year: ")
+        #self.fuel_level = 100
+        #self.mileage = 0
+
+    def show_cars_condition(self):
+        print(f"{self.brand}-{self.model} ({self.year}): distance->{self.mileage} miles, fuel level->{self.fuel_level}%")
+    
+    def __str__(self):
+        return f"{self.brand}-{self.model} ({self.year})"
+    
+    def enter_miles(self):
+        while True:
+            try: 
+                miles = int(input(f"How many miles did you drive with {self}? "))
+            except ValueError or TypeError:
+                print("Please enter a positive whole number.")
+                continue
+            if miles < 0:
+                print("Please enter a positive whole number.")
+            # fuel check
+            elif miles * 0.1 > self.fuel_level:
+                print(f"Not enough fuel in {self}. The fuel level is {self.fuel_level}%.")
+            else: break
+        return miles
+
+    def drive(self, miles):
+        self.mileage += miles
+        self.fuel_level -= miles * 0.1
+        print(f"{self} traveled {miles} miles.")
+        Car.total_distance += miles
+    
+    def refuel(self):
+        while True:
+            try: 
+                fueled_quantity = float(input(f"How many percent of fuel level do you want to refuel into {self}? "))
+            except ValueError or TypeError:
+                print("Please enter a positive whole number as percent. (except error)")
+                continue
+            if self.fuel_level + fueled_quantity > 100:
+                print(f"Is to much. You should only {100-self.fuel_level} refuel to full tank. (refuel error)")
+            elif fueled_quantity < 0.0:
+                print("Please enter a positive whole number. (negativ error)")
+            else: 
+                self.fuel_level += fueled_quantity
+                print(f"{self} is refuled: {self.fuel_level}")
+                break
+        return self.fuel_level 
+
+class Fleet(Car):
```

Comment:
Itt az öröklés nem szükséges. A Fleet osztálynak nincs szüksége semmire amit a Car tud. 
Nem akarod vezetni a flottát, nem akarod tankolni a flottát nincs üzemanyag szintje a flottának stb.
Itt a kapcsolat a composition-re utal, amit helyesen is csinálsz amikor a new_car_to_fleet()-et hívod meg.
A flottának vannak autói = composition
A flotta az egy autó = öröklés (ez ugye nem igaz)